### ISSUE TITLE
Hitting DEL during Save deletes the model / PC

### ISSUE THREAD

**Issue by rowild** [CloudCompare/CloudCompare#2076]
**Describe the bug**
Once a model / PointCloud is finished, I hit the Save button. In the input field that pops up I usually change the name. When I want to delete something of that string given using the delete key, the actual active model / PC gets deleted.

**How to reproduce**
Steps to reproduce the behaviour - include images if relevant:
1. Click on "Current Entity" icon in order to save it
2. In the popup dialog, delete letters using the DEL key

**Expected behaviour**
The model / PC should not be deleted

**Additional context**

**Your environment**
 - CC Version (or git hash/tag): 2.13.2
 - OS & Version: MacOS 14
 - Graphics card: Apple M2

**Comment by dgirardeau** [comment]
Hum, I can't reproduce this issue on Windows at least.

@prascle when you have some time, can you try to reproduce this issue maybe?

**Comment by rowild** [comment]
Thank you for your feedback, @dgirardeau ! – I am not sure if that helps, but here is a little video of whats happening on my Mac M2:
https://dl.rowild.at/cloudcompare-del-key-deletes-item.mp4

I also tested some websites that display the key code, the results are the same everywhere, here are screenshots from https://www.toptal.com/developers/keycode:

Backspace = keycode 8
![backspace_keycode_8](https://github.com/user-attachments/assets/ad5da468-97fe-4805-bf71-f44cee2a95f7)

Delete = keycode 46
![delete_keycode_46](https://github.com/user-attachments/assets/5cd55753-e9b0-4148-8b9a-4b4b4b8c4731)

The delete key is not part of the native MacBook keyboard, but comes from my external Logitech MX Keys keyboard. Maybe this is the root of my problem. (Logi provides a software to manipulate keyboards settings called "Logi Options+", but I have not changed any settings. It has the latest drivers and it is setup to mimick the Mac keyboard.)

**Comment by dgirardeau** [comment]
Thanks for the additional tests. I will need someone with a Mac (and ideally directly supporting on the macOS version of CloudCompare) to reproduce this issue.

The strange thing is that the File opening/saving dialogs are supposed to be 'modal' and should not transfer the keyboard events to any other component of the GUI. I guess this is different on macOS...

**Comment by prascle** [comment]
Hi, sorry for the very late reply.
I reproduce the bug on my mac M2 using the \<fn\> \<backspace\> combination which is delete (code 46)! 
With \<backspace\> only, the behavior is correct.
I'll try to correct this in the next binary beta...
Best regards,
Paul

**Comment by rowild** [comment]
@prascle Thank you! That would be great, if that could be fixed soon! I am losing so many edits during saving, simply because I want to change the name of the export by deleting the last couple letters. A common, muscle-memorized workflow... 

Would it be possible to change the status from "not_yet_confirmed" to "confirmed" (or whatever system you have)?

**Comment by rowild** [comment]
@dgirardeau @prascle Meanwhile this issue is confirmed, even though the status says differently. Any chance to tackle this problem?

**Comment by prascle** [comment]
@rowild I have a working fix for this bug on my [branch](https://github.com/CloudCompare/CloudCompare/compare/master...prascle:CloudCompare:macOSactionDelete). I released 2 alpha binaries of CloudCompare 2.14 today on my [site](https://www.simulation.openfields.fr/index.php/cloudcompare-downloads/7-cloudcompare-macos-binaries). There are probably still some dialogs that are not covered by this patch.

@dgirardeau I haven't found a generic solution to this problem (unless I permanently disable the delete shortcut key). I've only protected a large number of dialogs in which there's a reason to press <delete>.
So, no pull request until I have a better idea!

**Comment by prascle** [comment]
As a side effect of #2152, I've found that disabling native load/save dialogs (View menu / Display settings) solves the problem, without my fix above, and it works with versions 2.13.x.
